src/Database/PostgreSQL/PQTypes/Utils.hs

```diff
@@ -67,6 +71,30 @@ runQuery01 sql = do
 runQuery01_ :: (IsSQL sql, MonadDB m, MonadThrow m) => sql -> m ()
 runQuery01_ = void . runQuery01
 
+-- | Specialization of 'runQuery' that checks whether affected/returned
+-- number of rows is exactly 1.
+-- Otherwise, 'AffectedRowsMismatch' exception is thrown.
+{-# INLINABLE runQuery1_ #-}
+runQuery1_ :: (IsSQL sql, MonadDB m, MonadThrow m) => sql -> m ()
+runQuery1_ sql = do
+  n <- runQuery sql
+  when (n /= 1) $ throwDB AffectedRowsMismatch {
+    rowsExpected = [(1, 1)]
+  , rowsDelivered = n
+  }
+
+-- | Specialization of 'runQuery' that checks whether affected/returned
+-- number of rows is greater than 0.
+-- Otherwise, 'AffectedRowsMismatch' exception is thrown.
+{-# INLINABLE runQueryNot0_ #-}
+runQueryNot0_ :: (IsSQL sql, MonadDB m, MonadThrow m) => sql -> m ()
+runQueryNot0_ sql = do
+  n <- runQuery sql
+  when (n == 0) $ throwDB AffectedRowsMismatch {
+    rowsExpected = []
```

Comment:
You can use `(1, maxBound)` here, it's a bit weird but IMO better than empty range :thinking: